server/api/middlewares/authentication/authenticate.js

```diff
@@ -0,0 +1,78 @@
+import Sequelize from 'sequelize';
+import passport from 'passport';
+import { Strategy as JWTStrategy, ExtractJwt } from 'passport-jwt';
+import GoogleStrategy from 'passport-google-oauth20';
+import { Strategy as FacebookStrategy } from 'passport-facebook';
+import { Strategy as TwitterStrategy } from 'passport-twitter';
+import dotenv from 'dotenv';
+
+
+dotenv.config();
+
+const User = 'mock User';
+const { Op } = Sequelize;
+const options = {
+  secretOrKey: process.env.JWT_SECRET,
+  jwtFromRequest: ExtractJwt.fromAuthHeaderAsBearerToken(),
+};
+
+const {
+  GOOGLE_CLIENT_ID: googleClientID,
+  GOOGLE_CLIENT_SECRET: googleClientSecret,
+  FACEBOOK_APP_ID: facebookAppID,
+  FACEBOOK_APP_SECRET: facebookAppSecret,
+  TWITTER_CONSUMER_KEY: consumerKey,
+  TWITTER_CONSUMER_SECRET: consumerSecret,
+  DOMAIN: domain
+} = process.env;
+
+
+const credentials = (platform, clientID, clientSecret) => {
+  const callbackURL = `${domain}/api/user/${platform}/redirect`;
+  return platform !== 'twitter'
+    ? { clientID, clientSecret, callbackURL }
+    : { consumerKey, consumerSecret, callbackURL };
+};
+
+const passportCallback = (accessToken, refreshToken, {
+  id, username, displayName, name, emails: [{ value: email }], photos: [{ value: imageUrl }]
+}, done) => {
+  const { givenName: firstname = displayName, familyName: lastname } = name || {};
```

Comment:
@ifiokeyo Facebook and Google provide 'givenName' and 'familyName' in a 'name' object, while Twitter supplies just 'displayName' and there is no 'name' object. So I destructure the 'name' object if it exists(Facebook and Google), but if it doesn't, I destructure an empty object as default (to avoid javascript throwing an error) and set 'firstName' to Twitter's 'displayName'.